utils/statd/hostname.c

```diff
@@ -316,3 +329,49 @@ statd_matchhostname(const char *hostname1, const char *hostname2)
 			(result ? "matched" : "did not match"));
 	return result;
 }
+
+/**
+ * statd_matchhostname_pa - check if two hostnames are equivalent
+ * @hostname1: C string containing hostname
+ * @hostname2: C string containing hostname
+ *
+ * If either hostname is a presentation addresss then further
+ * comparisons beyond strcasecmp are not performed.
+ */
+_Bool
+statd_matchhostname_pa(const char *hostname1, const char *hostname2)
+{
+	char buf[INET_ADDRSTRLEN];
+	_Bool matched = false;
+
+	if (strcasecmp(hostname1, hostname2) == 0) {
+		matched = true;
+		goto out;
+	}
+
+	/*
+	 * Stop further matching if either hostname is an IPv4 network address.
+	 *
+	 * This is used during SM_MON/SM_UNMON requests to avoid stalls from
+	 * flaky DNS servers when locks are requested from clients that have
+	 * provided an ip address for the mon_name.
+	 */
+
+	if (inet_ntop(AF_INET, hostname1, buf, sizeof(buf)) != NULL) {
```

Comment:
I could be wrong here but I think we want to use inet_aton() to attempt convert the IPv4 hostname string.

inet_ntop() goes the other direction and expects the src/input to be a `struct in_addr`

Reply:
Great catch!  Somehow I got it backwards in my head.  Looks like `inet_aton ` is also now deprecated and `inet_pton` is the preferred  interface.